src/prefect/orion/orchestration/core_policy.py

```diff
@@ -406,13 +442,20 @@ class PreventTransitionsFromTerminalStates(BaseOrchestrationRule):
     FROM_STATES = TERMINAL_STATES
     TO_STATES = ALL_ORCHESTRATION_STATES
 
+    NAMED_STATE_EXCEPTIONS = {
+        "Restarting",
+        "Retrying",
+        "AwaitingRestart",
+    }
+
     async def before_transition(
         self,
         initial_state: Optional[states.State],
         proposed_state: Optional[states.State],
         context: OrchestrationContext,
     ) -> None:
-        await self.abort_transition(reason="This run has already terminated.")
+        if proposed_state.name not in self.NAMED_STATE_EXCEPTIONS:
+            await self.abort_transition(reason="This run has already terminated.")
```

Comment:
This makes me a bit uncomfortable. I guess I don't see a clear path around it at this time though.